var/spack/repos/builtin/packages/adol-c/package.py

```diff
@@ -46,39 +46,61 @@ class AdolC(AutotoolsPackage):
     variant('openmp',   default=False, description='Enable OpenMP support')
     variant('sparse',   default=False, description='Enable sparse drivers')
     variant('examples', default=True,  description='Install examples')
+    variant('boost',    default=False, description='Enable boost')
 
     # Build dependencies
     depends_on('automake', type='build', when='@develop')
     depends_on('autoconf', type='build', when='@develop')
     depends_on('libtool',  type='build', when='@develop')
     depends_on('m4',       type='build', when='@develop')
 
+    # Link dependencies
+    depends_on('boost+system', when='+boost')
```

Comment:
I don't see a `system` variant in `boost`

Reply:
> I don't see a system variant in boost

    system [on]              True, False       Compile with system library

p.s. it would not work if I use non-existing variant.

Reply:
Ah we autogenerate variants too, thanks